Sources/Swift/Tools/SentryViewPhotographer.swift

```diff
@@ -45,18 +45,24 @@ class SentryViewPhotographer: NSObject, SentryViewScreenshotProvider {
         dispatchQueue.dispatchAsync {
             let screenshot = UIGraphicsImageRenderer(size: imageSize, format: .init(for: .init(displayScale: 1))).image { context in
                 
+                let imageRect = CGRect(origin: .zero, size: imageSize)
                 context.cgContext.addRect(CGRect(origin: CGPoint.zero, size: imageSize))
                 context.cgContext.clip(using: .evenOdd)
                 UIColor.blue.setStroke()
                 
                 context.cgContext.interpolationQuality = .none
                 image.draw(at: .zero)
                 
+                var latestRegion: RedactRegion?
                 for region in redact {
                     let rect = CGRect(origin: CGPoint.zero, size: region.size)
                     var transform = region.transform
                     let path = CGPath(rect: rect, transform: &transform)
                     
+                    defer { latestRegion = region }
+                    
+                    guard region != latestRegion && imageRect.contains(path.boundingBoxOfPath) else { continue }
```

Comment:
`m`: It seems like this speeds up the performance significantly. Maybe it's worth adding a comment for this, cause we don't have a test to validate that we do this.

Reply:
We do have a test to validate this.
Its the only test added to this PR